extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -447,8 +447,10 @@ export class NextPageService {
 
     const urlService = Services.urlForDoc(dev().assertElement(this.element_));
     const {title} = documentRef.amp;
-    const {pathname} = urlService.parse(documentRef.ampUrl);
-    this.win_.history.replaceState({}, title, pathname);
+    const url = urlService.parse(documentRef.ampUrl);
+    // Strip the origin from the URL to get the path with query string.
+    const fullPath = documentRef.ampUrl.substr(url.origin.length);
```

Comment:
This may not work with IDN, origin may be the Punny code. I don't know if `documentRef.ampUrl` would already be in punny code or not. 

Might be safer to just do `pathname + search`

Reply:
Good spot! `documentRef.ampUrl` is user supplied, so may or may not be encoded. The parsed version will always be punycode, so yeah this won't work. Updated.